bio_pieces/parallel_blast.py

```diff
@@ -73,23 +71,39 @@ def parallel_blast(inputfile, outfile, ninst, db, blasttype, task, blastoptions)
         None if blastx/blastp
     :param str blastoptions: other options to pass to blast
     '''
+    if has_duplicate_args(blastoptions, STATIC_BLAST_ARGS):
```

Comment:
blastoptions is a string so for both your cases you would need to convert blastoptions to a list using shlex.split.

```
set(STATIC_BLAST_ARGS).intersection(shlex.split(blastoptions))
```

Typically I think this is frowned upon as you have too many things going on in one line making it hard to read an non-pythonic.

I'll remove the function though because it doesn't really have a name that tells what it is doing and it removes a test class though.